Add tutorial of free flow over porous media (#678)
precice/tutorials: free-flow-over-porous-media/README.md

---
title: Free flow over porous media 2D
permalink: tutorials-free-flow-over-porous-media-2d.html
keywords: DuMux, porous media
summary: Flow-flow coupling example with porous media field and free flow field.
---

{% note %}
Get the [case files of this tutorial](https://github.com/precice/tutorials/tree/master/free-flow-over-porous-media-2d). Read how in the [tutorials introduction](https://precice.org/tutorials.html).
{% endnote %}

## Setup

> This tutorial solves a simple coupled system consisting of a one-phase free flow and a one-phase flow in a porous media.

Reviewer: ```suggestion
This tutorial solves a coupled system consisting of a one-phase free flow and a one-phase flow in a porous media.
```

> A pressure gradient is applied to the free flow domain from left to right, while at the top of the free-flow we have a non-permeable wall with no-slip boundary conditions. In the porous media, we assume no-flow across the domain boundaries (left, bottom and right boundary). At the interface we assume a no-slip condition.

Reviewer: ```suggestion
A pressure gradient is applied to the free flow domain from left to right. The top edge of the free-flow is a non-permeable wall with no-slip boundary conditions. In the porous media, there is a no-flow condition across the domain boundaries (left, bottom, and right boundaries). At the interface, a no-slip condition applies. The case is stationary (solved to a steady-state solution).
```

The setting is illustrated in the following figure:

![Free flow over porous media setup](images/tutorials-free-flow-over-porous-media-setup.png)

## Configuration

preCICE configuration (image generated using the [precice-config-visualizer](https://precice.org/tooling-config-visualization.html)):

![preCICE configuration visualization](images/tutorials-free-flow-over-porous-media-precice-config-visualization.png)

## Available solvers

> Both the flow in free flow and porous media can be solved using the simulation framework [DuMu<sup>x</sup>](https://git.iws.uni-stuttgart.de/dumux-repositories/dumux/).

Reviewer: ```suggestion
Both the participants are computed using the simulation code [DuMu<sup>x</sup>](https://git.iws.uni-stuttgart.de/dumux-repositories/dumux/).
```

## Solver setup

To solve the flows with the DuMux framework, the necessary DUNE modules need to be downloaded and set up. This is done by running `sh setup-dumux.sh` in the tutorial folder.

> Note that if an existing installation of DUNE modules is detected in a default location, this may lead to problems in running the `setup-dumux.sh` script. The environment variable DUNE_CONTROL_PATH is suppressed by the script.

Reviewer: ```suggestion
Note that if an existing installation of DUNE modules is detected in a default location, this may lead to problems in running the `setup-dumux.sh` script. The script suppresses the environment variable `DUNE_CONTROL_PATH`.
```

> Note that if an existing installation of DUNE modules is detected in a default location, this may lead to problems in running the `setup-dumux.sh` script. The script suppresses the environment variable `DUNE_CONTROL_PATH`.

Reviewer 2: Another reason to offload the installation to the user.
Is there a particular reason to do something special here?

Reviewer: Yes, I find this note valuable, for reasons mentioned above.

> To recompile only the simulation, run `sh compile-dumux-cases.sh` in the tutorial folder.

Reviewer: ```suggestion
To only recompile the participants, run `sh compile-dumux-cases.sh` in the tutorial folder.
```

## Running the simulation

> You can find the corresponding `run.sh`script for running the case in the folders corresponding to the solvers you want to use.

Reviewer: ```suggestion
Each participant has a `run.sh` script.
```

To run the free-flow participant, run:

```bash
cd free-flow-dumux
./run.sh
```

To run the porous-media participant, run:

```bash
cd porous-media-dumux
./run.sh
```

> So far only serial execution is supported for this tutorial, which takes less than two minutes to finish.

Reviewer: ```suggestion
Participants can be executed only in serial. Parallel execution is not supported. The case takes approximately two minutes to finish.
```

## Post-processing

> The VTU files from both solvers could be rendered and inspected with ParaView.

Reviewer: ```suggestion
Both participants create export VTU files, which can be viewed using ParaView.
```

## Further information

> The simulation is stationary and the results of the pressure and the velocity fields are as follows:

Reviewer: ```suggestion
The results of the pressure and the velocity fields are as follows:
```
